src/core/util.ts

```diff
@@ -653,20 +653,67 @@ export function isPrimitive(obj: any): boolean {
     return obj[primitiveKey];
 }
 
+interface MapInterface<T, KEY extends string | number = string | number> {
+    delete(key: KEY): boolean;
+    get(key: KEY): T | undefined
+    set(key: KEY, value: T): this
+    keys(): KEY[];
+    forEach(callback: (value: T, key: KEY) => void): void
+}
+
+class MapPolyfill<T, KEY extends string | number = string | number> implements MapInterface<T, KEY> {
+    private data: Record<KEY, T> = {} as Record<KEY, T>;
+
+    delete(key: KEY): boolean {
+        const existed = this.data.hasOwnProperty(key);
+        if (existed) {
+            delete this.data[key];
+        }
+        return existed;
+    }
+    get(key: KEY): T | undefined {
+        return this.data[key];
+    }
+    set(key: KEY, value: T): this {
+        this.data[key] = value;
+        return this;
+    }
+    keys(): KEY[] {
+        return keys(this.data);
+    }
+    forEach(callback: (value: T, key: KEY) => void): void {
+        // This is a potential performance bottleneck, see details in
+        // https://github.com/ecomfe/zrender/issues/965, however it is now
+        // less likely to occur as we default to native maps when possible.
+        for (const key in this.data) {
```

Comment:
Here we can cache `this.data` to a constant.

```ts
const data = this.data;
for (const key in data) {
    if (data.hasOwnProperty(key)) {
        callback(data[key], key);
    }
}
```

Reply:
I would not expect this to have a noticeable performance impact on performance as we are just referencing to a prototype chain, but I ran a quick micro benchmark to confirm (https://jsbench.me/3vl9wtiswy/1). I would prefer to go for simple code here, but if you have a strong preference please let me know and I will change it.

Reply:
Oh I see. If you have a strong preference then I wouldn't mind changing it, but I don't think it makes much difference in bundle size

Reply:
It can only reduce several bytes after minification insignificant but we can do it.